src/utils.ts

```diff
@@ -28,3 +29,66 @@ export function paginateStrategies(space, network, strategies) {
     return strategy;
   });
 }
+
+const networks = {
+  '1': 7929876,
+  '137': 9834491,
+  '100': 4108192
+};
```

Comment:
Maybe we can call a subgraph like in multichain right? 

Also, did you consider https://github.com/monosux/ethereum-block-by-date ? it is doing the same calculation in the background

Reply:
Didnt see this lib, but looking on it i believe it doesnt send request in parralel and would most likely take longer to resolve